src/Core/Notifications/INotificationService.cs

```diff
@@ -18,10 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using System.Linq;
+using SonarLint.VisualStudio.Core.InfoBar;
+
 namespace SonarLint.VisualStudio.Core.Notifications
 {
-    public interface INotificationService
+    public interface INotificationService : IDisposable
     {
         void ShowNotification(INotification notification);
     }
+    
+    public sealed class NotificationService : INotificationService
+    {
+        /// <summary>
+        /// Taken from "ToolWindowGuids80.ErrorList"
+        /// </summary>
+        internal static readonly Guid ErrorListToolWindowGuid = new Guid("D78612C7-9962-4B83-95D9-268046DAD23A");
+
+        private readonly IInfoBarManager infoBarManager;
+        private readonly IThreadHandling threadHandling;
+
+        private Tuple<IInfoBar, INotification> activeNotification;
+
+        public NotificationService(IInfoBarManager infoBarManager, IThreadHandling threadHandling)
+        {
+            this.infoBarManager = infoBarManager;
+            this.threadHandling = threadHandling;
+        }
+
+        public void ShowNotification(INotification notification)
+        {
+            if (notification == null)
+            {
+                throw new ArgumentNullException(nameof(notification));
+            }
+            // todo: check if already shown
+
+            threadHandling.RunOnUIThread(() =>
+            {
+                RemoveExistingInfoBar();
+
+                var buttonTexts = notification.Actions.Select(x => x.CommandText).ToArray();
+
+                var infoBar = infoBarManager.AttachInfoBarWithButtons(ErrorListToolWindowGuid,
+                    notification.Message,
+                    buttonTexts,
+                    SonarLintImageMoniker.OfficialSonarLintMoniker);
+
+                activeNotification = new Tuple<IInfoBar, INotification>(infoBar, notification); 
+                activeNotification.Item1.ButtonClick += CurrentInfoBar_ButtonClick;
+                activeNotification.Item1.Closed += CurrentInfoBar_Closed;
+            });
+        }
+
+        private void CurrentInfoBar_ButtonClick(object sender, InfoBarButtonClickedEventArgs e)
+        {
+            var matchingAction = activeNotification.Item2.Actions.FirstOrDefault(x => x.CommandText == e.ClickedButtonText);
```

Comment:
This will be on the UI thread, won't it?
If so, we should guard against unhandled exceptions so we don't crash VS.

The same applies to the "closed" handler, but it's less likely to throw since it's not performing an arbitrary task (the existing infobar code doesn't have error handling for the close).